recipes/libcurl/all/conanfile.py

```diff
@@ -103,7 +111,9 @@ def configure(self):
 
         # Deprecated options
         # ===============================
-        self.output.warn("with_openssl, with_winssl, darwin_ssl and with_wolfssl options are deprecated. Use with_ssl option instead.")
+        if (tools.is_apple_os(self.settings.os) and (self.options.with_openssl or self.options.with_wolfssl or self.options.with_winssl or not self.options.darwin_ssl)) or \
```

Comment:
Another way to detect use of these deprecated options, would be to give them an additional option, e.g.: `[True, False, "deprecated"]` and default option value `deprecated`.
If one of these option is set either to `True` or `False`, then it can print the warning.
This way, the test in `configure` does not depends on the current os.

Reply:
yes, good suggestion, I'll try yo add "deprecated"

Reply:
https://github.com/conan-io/conan-center-index/pull/3209/commits/f2849a4b02311737310d60311fd77619918b9e76